### ISSUE TITLE
Unable to run Civ V

### ISSUE THREAD

**Issue by AngrySaltyFish** [ValveSoftware/steam-for-linux#5734]
#### system information

My system is fully up to date when this has been posted 

Computer Information:
    Manufacturer:  System76
    Form Factor: Laptop
    No Touch Input Detected

Processor Information:
    CPU Vendor:  GenuineIntel
    CPU Brand:  Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    CPU Family:  0x6
    CPU Model:  0x9e
    CPU Stepping:  0x9
    CPU Type:  0x0
    Speed:  3800 Mhz
    8 logical processors
    4 physical processors
    HyperThreading:  Supported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported
    AES:  Supported
    AVX:  Supported
    CMPXCHG16B:  Supported
    LAHF/SAHF:  Supported
    PrefetchW:  Unsupported

Operating System Version:
    Linux 4.x (64 bit)
    Kernel Name:  Linux
    Kernel Version:  4.18.5-arch1-1-ARCH
    X Server Vendor:  The X.Org Foundation
    X Server Release:  12001000
    X Window Manager:  bspwm
    Steam Runtime Version:  steam-runtime-beta-release_2018-06-14

Video Card:
    Driver:  NVIDIA Corporation GeForce GTX 1070/PCIe/SSE2
    Driver Version:  4.6.0 NVIDIA 396.54
    OpenGL Version: 4.6
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 75 Hz
    VendorID:  0x10de
    DeviceID:  0x1be1
    Revision Not Detected
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    Primary Display Resolution:  1920 x 1080
    Desktop Resolution: 1920 x 1080
    Primary Display Size: 15.04" x 8.46" (17.24" diag)
                                            38.2cm x 21.5cm (43.8cm diag)
    Primary Bus: PCI Express 16x
    Primary VRAM: 8192 MB
    Supported MSAA Modes:  2x 4x 8x 16x 

Sound card:
    Audio device: Realtek ALC898

Memory:
    RAM:  15993 Mb

Miscellaneous:
    UI Language:  English
    LANG:  en_GB.UTF-8
    Total Hard Disk Space Available:  938771 Mb
    Largest Free Hard Disk Block:  430705 Mb
    VR Headset: None detected

#### Please describe your issue in as much detail as possible:

Initially steam booted up as expected and I attempted to run the game "Sid Meier's Civilization® V" 
(https://store.steampowered.com/app/8930/Sid_Meiers_Civilization_V/)

Unfortunatly Civ V almost instantly crashes when booting so I ran it from the terminal and quickly discovered some unusually error messages. I went on the internet and found people with similar setups with the same problem and applied the fix. The fix is descirbed at the two following links:

https://forum.manjaro.org/t/civ-v-wont-launch-after-update/10825
https://steamcommunity.com/app/8930/discussions/1/364042703858680804/
https://bbs.archlinux.org/viewtopic.php?id=222265

I opted for the neater option by adding the following to the launch options of civ 5:

 LD_PRELOAD='/usr/$LIB/libgcc_s.so.1 /usr/$LIB/libxcb.so.1 /usr/$LIB/libgpg-error.so ./libcxxrt.so /usr/lib32/libstdc++.so.6' %command%

This made the unsual error messages disappear but civ will stil not run. I then attempted to modifiy the fix which had be suggested in some of the artiticles describing the fix by executing "STEAM_RUNTIME=0" and then trying to run the application from the terminal.

This made things worse as with this option steam will not boot and the following text is left in the terminal:

```
/mnt/externalHD/home/james/.local/share/Steam/steam.sh: line 106: VERSION_ID: unbound variable
/mnt/externalHD/home/james/.local/share/Steam/steam.sh: line 106: VERSION_ID: unbound variable
Running Steam on arch  64-bit
STEAM_RUNTIME is disabled by the user
Installing breakpad exception handler for appid(steam)/version(1535576546)
Installing breakpad exception handler for appid(steam)/version(1535576546)
Installing breakpad exception handler for appid(steam)/version(1535576546)
[0905/141030.627609:WARNING:crash_reporting.cc(254)] Failed to set crash key: UserID with value: 0
[0905/141030.627658:WARNING:crash_reporting.cc(254)] Failed to set crash key: BuildID with value: 1535573251
[0905/141030.627680:WARNING:crash_reporting.cc(254)] Failed to set crash key: SteamUniverse with value: Public
[0905/141030.627683:WARNING:crash_reporting.cc(254)] Failed to set crash key: Version with value: 1535573251
[0905/141030.627686:WARNING:crash_reporting.cc(254)] Failed to set crash key: ProductName with value: cefwebhelper
[0905/141030.627689:WARNING:crash_reporting.cc(254)] Failed to set crash key: Vendor with value: Valve
[0905/141030.634234:ERROR:gpu_process_transport_factory.cc(1029)] Lost UI shared context.
html_chrome.cpp (394) : Assertion Failed: CefCrashReportingEnabled()
[0905/141030.658267:WARNING:crash_reporting.cc(254)] Failed to set crash key: SteamUniverse with value: Dev
[0905/141030.658882:WARNING:crash_reporting.cc(254)] Failed to set crash key: UserID with value: 0
[0905/141030.658901:WARNING:crash_reporting.cc(254)] Failed to set crash key: BuildID with value: 1535576546
[0905/141030.658910:WARNING:crash_reporting.cc(254)] Failed to set crash key: SteamUniverse with value: Public
[0905/141030.658918:WARNING:crash_reporting.cc(254)] Failed to set crash key: Version with value: 1535576546
[0905/141030.658926:WARNING:crash_reporting.cc(254)] Failed to set crash key: ProductName with value: cefwebhelper
[0905/141030.658934:WARNING:crash_reporting.cc(254)] Failed to set crash key: Vendor with value: Valve
CApplicationManagerPopulateThread took 3 milliseconds to initialize (will have waited on CAppInfoCacheReadFromDiskThread)
Installing breakpad exception handler for appid(steam)/version(1535576546)
CAppInfoCacheReadFromDiskThread took 32 milliseconds to initialize
Found unsupported CDM version 1.4.9.1088, using last known good version 1.4.8.1008
Generating new string page texture 2: 48x256, total string texture memory is 49.15 KB
Generating new string page texture 3: 384x256, total string texture memory is 442.37 KB
(steam:26867): Gtk-WARNING **: 14:10:41.623: gtk_disable_setlocale() must be called before gtk_init()
WARNING: setlocale('en_US.UTF-8') failed, using locale: 'C'. International characters may not work.
[2018-09-05 14:10:30] Startup - updater built Aug 29 2018 20:06:41
[2018-09-05 14:10:30] Verifying installation...
[2018-09-05 14:10:30] Verification complete
roaming config store loaded successfully - 734 bytes.
migrating temporary roaming config store
appdatacache.cpp (2124) : Assertion Failed: !bSharedKVSymbols
crash_20180905141043_19.dmp[26944]: Uploading dump (out-of-process)
/tmp/dumps/crash_20180905141043_19.dmp
appdatacache.cpp (2124) : Assertion Failed: !bSharedKVSymbols
appdatacache.cpp (2124) : Assertion Failed: !bSharedKVSymbols
crash_20180905141043_19.dmp[26944]: Finished uploading minidump (out-of-process): success = yes
crash_20180905141043_19.dmp[26944]: response: CrashID=bp-2d3caf6c-4bae-4ebd-9009-5c0902180905
crash_20180905141043_19.dmp[26944]: file ''/tmp/dumps/crash_20180905141043_19.dmp'', upload yes: ''CrashID=bp-2d3caf6c-4bae-4ebd-9009-5c0902180905''
```
There appears to be a crash log saved to the tmp directory but I am unsure on how to exaime it and how to give it to you via github. If you can suggest a means of giving it to you i will happily comply.

Interestingly if I enable steam runtime then steam boots up as expected but civ 5 does not work.

#### Steps for reproducing this issue:

1. Have Civ 5 installed via the steam client
2. Running the game at this point should crash and give errors like ```/Civ5XP: Symbol `_ZTVN10__cxxabiv120__si_class_type_infoE' has different size in shared object, consider re-linking``` when running from the terminal
3. Attempt to apply a fix by adding the following to the launch options " LD_PRELOAD='/usr/$LIB/libgcc_s.so.1 /usr/$LIB/libxcb.so.1 /usr/$LIB/libgpg-error.so ./libcxxrt.so /usr/lib32/libstdc++.so.6' %command%"
4. Civ 5 is still broken so attempt the suggested modification by setting STEAM_RUNTIME=0
5. Steam crashes at boot.

I am very unfamilar to the internel workings of steam so the fix may be something very simple, if so apologies for wasting your time.

If you have any queries do not hesitate to ask

**Comment by kisak-valve** [comment]
Hello @Archangel7777777, if you want to try to run Steam on Arch without the steam runtime, it'd be a good idea to make sure you have the [steam-native-runtime](https://www.archlinux.org/packages/multilib/x86_64/steam-native-runtime/) package and launch steam with `steam-native` instead of using `STEAM_RUNTIME=0 steam`

**Comment by Archangel7777777** [comment]
So I had a go at running the `steam-native` package and found that it crash soon after logging in. The output produced by the terminal is very similar to the one produced by in the original post.
Hope this helps let me know if you need anything else.

**Comment by Archangel7777777** [comment]
So I managed to get `steam-native` up and running by reading up on it at the arch wiki:https://wiki.archlinux.org/index.php/Steam/Troubleshooting#Steam_native_runtime and realising I needed to install the `steam-native-runtime` package. Unfortunatly civ 5 is still not working. Running it in the terminal gave this output

```
/mnt/externalHD/home/james/.local/share/Steam/steam.sh: line 106: VERSION_ID: unbound variable
/mnt/externalHD/home/james/.local/share/Steam/steam.sh: line 106: VERSION_ID: unbound variable
Running Steam on arch  64-bit
STEAM_RUNTIME is disabled by the user
Installing breakpad exception handler for appid(steam)/version(1535576546)
Installing breakpad exception handler for appid(steam)/version(1535576546)
Installing breakpad exception handler for appid(steam)/version(1535576546)
[0905/164129.431710:WARNING:crash_reporting.cc(254)] Failed to set crash key: UserID with value: 0
[0905/164129.431763:WARNING:crash_reporting.cc(254)] Failed to set crash key: BuildID with value: 1535573251
[0905/164129.431785:WARNING:crash_reporting.cc(254)] Failed to set crash key: SteamUniverse with value: Public
[0905/164129.431789:WARNING:crash_reporting.cc(254)] Failed to set crash key: Version with value: 1535573251
[0905/164129.431792:WARNING:crash_reporting.cc(254)] Failed to set crash key: ProductName with value: cefwebhelper
[0905/164129.431795:WARNING:crash_reporting.cc(254)] Failed to set crash key: Vendor with value: Valve
[0905/164129.439990:ERROR:gpu_process_transport_factory.cc(1029)] Lost UI shared context.
html_chrome.cpp (394) : Assertion Failed: CefCrashReportingEnabled()
[0905/164129.449048:WARNING:crash_reporting.cc(254)] Failed to set crash key: SteamUniverse with value: Dev
[0905/164129.449657:WARNING:crash_reporting.cc(254)] Failed to set crash key: UserID with value: 0
[0905/164129.449677:WARNING:crash_reporting.cc(254)] Failed to set crash key: BuildID with value: 1535576546
[0905/164129.449687:WARNING:crash_reporting.cc(254)] Failed to set crash key: SteamUniverse with value: Public
[0905/164129.449695:WARNING:crash_reporting.cc(254)] Failed to set crash key: Version with value: 1535576546
[0905/164129.449704:WARNING:crash_reporting.cc(254)] Failed to set crash key: ProductName with value: cefwebhelper
[0905/164129.449712:WARNING:crash_reporting.cc(254)] Failed to set crash key: Vendor with value: Valve
Installing breakpad exception handler for appid(steam)/version(1535576546)
Installing breakpad exception handler for appid(steam)/version(1535576546)
Installing breakpad exception handler for appid(steam)/version(1535576546)
Installing breakpad exception handler for appid(steam)/version(1535576546)
Installing breakpad exception handler for appid(steam)/version(1535576546)
Installing breakpad exception handler for appid(steam)/version(1535576546)
CApplicationManagerPopulateThread took 3 milliseconds to initialize (will have waited on CAppInfoCacheReadFromDiskThread)
Installing breakpad exception handler for appid(steam)/version(1535576546)
Installing breakpad exception handler for appid(steam)/version(1535576546)
Installing breakpad exception handler for appid(steam)/version(1535576546)
Installing breakpad exception handler for appid(steam)/version(1535576546)
Installing breakpad exception handler for appid(steam)/version(1535576546)
Installing breakpad exception handler for appid(steam)/version(1535576546)
Installing breakpad exception handler for appid(steam)/version(1535576546)
Installing breakpad exception handler for appid(steam)/version(1535576546)
CAppInfoCacheReadFromDiskThread took 35 milliseconds to initialize
Installing breakpad exception handler for appid(steam)/version(1535576546)
Installing breakpad exception handler for appid(steam)/version(1535576546)
Found unsupported CDM version 1.4.9.1088, using last known good version 1.4.8.1008
Generating new string page texture 2: 48x256, total string texture memory is 49.15 KB
Generating new string page texture 3: 384x256, total string texture memory is 442.37 KB
Installing breakpad exception handler for appid(steam)/version(1535576546)

(steam:5709): Gtk-WARNING **: 16:41:40.415: gtk_disable_setlocale() must be called before gtk_init()
Installing breakpad exception handler for appid(steam)/version(1535576546)
roaming config store loaded successfully - 734 bytes.
migrating temporary roaming config store
[0905/164142.714780:WARNING:crash_reporting.cc(254)] Failed to set crash key: UserID with value: 0
[0905/164142.714846:WARNING:crash_reporting.cc(254)] Failed to set crash key: BuildID with value: 1535573251
[0905/164142.714850:WARNING:crash_reporting.cc(254)] Failed to set crash key: SteamUniverse with value: Public
[0905/164142.714853:WARNING:crash_reporting.cc(254)] Failed to set crash key: Version with value: 1535573251
[0905/164142.714856:WARNING:crash_reporting.cc(254)] Failed to set crash key: ProductName with value: cefwebhelper
[0905/164142.714860:WARNING:crash_reporting.cc(254)] Failed to set crash key: Vendor with value: Valve
Failed to init SteamVR because it isn't installed
ExecCommandLine: "'/mnt/externalHD/home/james/.local/share/Steam/ubuntu12_32/steam'"
System startup time: 14.33 seconds
Generating new string page texture 77: 128x256, total string texture memory is 573.44 KB
Generating new string page texture 78: 128x256, total string texture memory is 131.07 KB
Generating new string page texture 79: 256x256, total string texture memory is 835.58 KB
Generating new string page texture 80: 64x256, total string texture memory is 901.12 KB
Generating new string page texture 82: 32x256, total string texture memory is 933.89 KB
JS method call WebChat.GetWebChatLanguage with 1 arguments
JS method call WebChat.RegisterForFriendPostMessage with 1 arguments
JS method call WebChat.GetSignIntoFriendsOnStart with 1 arguments
JS method call WebChat.GetCurrentUserAccountID with 1 arguments
JS method call WebChat.RegisterForFriendPostMessage with 1 arguments
JS method call WebChat.GetWebChatURL with 1 arguments
JS method call FriendSettings.RegisterForSettingsChanges with 1 arguments
JS method call WebChat.RegisterForPushToTalkStateChange with 1 arguments
JS method call WebChat.GetPushToTalkEnabled with 1 arguments
JS method call WebChat.RegisterForFriendPostMessage with 1 arguments
JS method call WebChat.RegisterForComputerActiveStateChange with 1 arguments
JS method call SharedConnection.AllocateSharedConnection with 1 arguments
JS method call WebChat.RegisterOverlayChatBrowserInfoChanged with 1 arguments
JS method call FriendSettings.GetEnabledFeatures with 1 arguments
JS method call SharedConnection.SubscribeToEMsg with 2 arguments
JS method call SharedConnection.SubscribeToEMsg with 2 arguments
JS method call SharedConnection.SubscribeToEMsg with 2 arguments
JS method call SharedConnection.SubscribeToEMsg with 2 arguments
JS method call SharedConnection.SubscribeToEMsg with 2 arguments
JS method call SharedConnection.SubscribeToEMsg with 2 arguments
JS method call SharedConnection.SubscribeToEMsg with 2 arguments
JS method call SharedConnection.SubscribeToEMsg with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.SubscribeToClientServiceMethod with 2 arguments
JS method call SharedConnection.RegisterOnLogonInfoChanged with 2 arguments
JS method call SharedConnection.SendMsgAndAwaitResponse with 3 arguments
JS method call SharedConnection.SendMsgAndAwaitResponse with 3 arguments
JS method call SharedConnection.SendMsgAndAwaitResponse with 3 arguments
JS method call WebChat.GetOverlayChatBrowserInfo with 1 arguments
JS method call Parental.RegisterForParentalSettingsChanges with 1 arguments
JS method call Broadcast.RegisterForBroadcastStatus with 1 arguments
JS method call Broadcast.RegisterForViewerRequests with 1 arguments
JS method call SharedConnection.RegisterOnMessageReceived with 2 arguments
JS method call SharedConnection.SendMsg with 2 arguments
Installing breakpad exception handler for appid(steam)/version(1535576546)
JS method call SharedConnection.SendMsgAndAwaitResponse with 3 arguments
JS method call SharedConnection.SendMsgAndAwaitResponse with 3 arguments
JS method call SharedConnection.SendMsgAndAwaitResponse with 3 arguments
JS method call SharedConnection.SendMsgAndAwaitResponse with 3 arguments
JS method call SharedConnection.SendMsg with 2 arguments
JS method call SharedConnection.SendMsgAndAwaitResponse with 3 arguments
JS method call SharedConnection.SendMsgAndAwaitResponse with 3 arguments
JS method call SharedConnection.SendMsg with 2 arguments
JS method call WebChat.BSuppressPopupsInRestore with 1 arguments
Generating new string page texture 98: 256x256, total string texture memory is 393.22 KB
Generating new string page texture 105: 128x256, total string texture memory is 1.06 MB
Generating new string page texture 106: 512x256, total string texture memory is 1.59 MB
Generating new string page texture 108: 128x256, total string texture memory is 1.72 MB
Generating new string page texture 109: 256x256, total string texture memory is 1.98 MB
Generating new string page texture 131: 48x256, total string texture memory is 2.03 MB
GameAction [AppID 8930, ActionID 1] : LaunchApp changed task to ProcessingInstallScript with ""
GameAction [AppID 8930, ActionID 1] : LaunchApp changed task to SynchronizingCloud with ""
GameAction [AppID 8930, ActionID 1] : LaunchApp changed task to ProcessingShaderCache with ""
GameAction [AppID 8930, ActionID 1] : LaunchApp changed task to SiteLicenseSeatCheckout with ""
GameAction [AppID 8930, ActionID 1] : LaunchApp changed task to CreatingProcess with ""
GameAction [AppID 8930, ActionID 1] : LaunchApp waiting for user response to CreatingProcess ""
GameAction [AppID 8930, ActionID 1] : LaunchApp continues with user response "CreatingProcess"
Opted-in Controller Mask: 70
Game update: AppID 8930 "", ProcID 5875, IP 0.0.0.0:0
>>> Adding process 5875 for game ID 8930
GameAction [AppID 8930, ActionID 1] : LaunchApp changed task to WaitingGameWindow with ""
ERROR: ld.so: object '/mnt/externalHD/home/james/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
GameAction [AppID 8930, ActionID 1] : LaunchApp changed task to Completed with ""
>>> Adding process 5876 for game ID 8930
Game update: AppID 8930 "", ProcID 5876, IP 0.0.0.0:0
Setting breakpad minidump AppID = 8930
Steam_SetMinidumpSteamID:  Caching Steam ID:  76561198126894240 [API loaded no]
Game removed: AppID 8930 "", ProcID 5876 
No cached sticky mapping in ActivateActionSet.html_chrome.cpp (394) : Assertion Failed: CefCrashReportingEnabled()
Exiting workitem thread
WARNING: setlocale('en_US.UTF-8') failed, using locale: 'C'. International characters may not work.
[2018-09-05 16:41:28] Startup - updater built Aug 29 2018 20:06:41
[2018-09-05 16:41:28] Verifying installation...
[2018-09-05 16:41:28] Verification complete
[2018-09-05 16:42:02] Shutdown
```

**Comment by Archangel7777777** [comment]
@kisak-valve do you have any ideas on how to fix this?

**Comment by shumvgolove** [comment]
I had the same problem literally yesterday and today I was managed to solve it.

What I've done:

1. Completely deleted the folder `~ /.local/share/Steam` (all games will be deleted if they were installed there, so be careful).
2. Deleted packages along with the dependencies `pacman -Rs steam-native-runtime steam`.
3. Installed Steam `pacman -S steam`.
4. Set parameters for launching the game on Steam `LD_PRELOAD='./libcxxrt.so:/usr/$LIB/libstdc++.so.6' %command%` as was discribed [here ](https://wiki.archlinux.org/index.php/Steam/Game-specific_troubleshooting#Civilization_V).
5. The game started.

**Comment by AngrySaltyFish** [comment]
It's been a long time since I posted this issue have now found the solution so I thought I should post just in case someone else has the same issue and need some help. The solution I use to fix it is detailed in the gentoo wiki under https://wiki.gentoo.org/wiki/Steam/Games_troubleshooting#Sid_Meier.27s_Civilization_V. I had to recompile the packages described by the second bullet point with the correct options enabled.
